costmap_cspace_rviz_plugins/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+cmake_minimum_required(VERSION 3.1.3)
+project(costmap_cspace_rviz_plugins)
+
+set(PACKAGE_DEPENDS
+  roscpp
+  rviz
+  costmap_cspace_msgs
+)
+
+find_package(catkin REQUIRED COMPONENTS ${PACKAGE_DEPENDS})
+
+catkin_package(CATKIN_DEPENDS ${PACKAGE_DEPENDS})
+
+set(CMAKE_CXX_STANDARD 14)
+add_compile_options(-Wall)
+
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+)
+link_directories(${catkin_LIBRARY_DIRS})
+
+set(CMAKE_AUTOMOC ON)
+
+if(rviz_QT_VERSION VERSION_LESS "5")
+  message(STATUS "Using Qt4 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
+  find_package(Qt4 ${rviz_QT_VERSION} EXACT REQUIRED QtCore QtGui)
+  include(${QT_USE_FILE})
+else()
+  message(STATUS "Using Qt5 based on the rviz_QT_VERSION: ${rviz_QT_VERSION}")
+  find_package(Qt5 ${rviz_QT_VERSION} EXACT REQUIRED Core Widgets)
+  set(QT_LIBRARIES Qt5::Widgets)
+endif()
+
+add_definitions(-DQT_NO_KEYWORDS)
+
+set(SRC_FILES
+  include/costmap_cspace_rviz_plugins/c_space_3d_display.h
```

Comment:
I'm not very sure this is required.
I think cmake automatically detects dependencies to static header files.

Reply:
It is needed to pass `c_space_3d_display.h` to `add_library`. It seems that moc of QT builds additional C++ source file from  headers.
I added `HEADER_FILES` variable to split headers from sources.

Reply:
Ah yes, generated files aren't automatically detected by cmake.
In trajectory_tracker_rviz_plugins, I used `qtN_wrap_cpp` to explicitly handle generated cpp files.
https://github.com/at-wat/neonavigation_rviz_plugins/blob/75a9759cbc7057fe62198b00751af15a74b17713/trajectory_tracker_rviz_plugins/CMakeLists.txt#L26-L52

Reply:
Changed to use qt4/qt5_wrap_cpp.